src/validation/transaction.cairo

```diff
@@ -30,20 +30,28 @@ pub fn validate_transaction(
     //      - https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp
 
     //      - Coinbase is mature (if some input spends coinbase tx from the past)
-    let first_input = *tx.inputs[0];
-    if first_input.previous_output.is_coinbase {
-        let coinbase_block_height = first_input.previous_output.block_height;
-        if block_height <= coinbase_block_height + 100 {
-            return Result::Err(
-                format!(
-                    "[validate_transaction] coinbase transaction not mature (current height: {}, coinbase height: {})",
-                    block_height,
-                    coinbase_block_height
-                )
-                    .into()
-            );
-        }
-    }
+    let mut errors = ArrayTrait::new();
+
+    for input in *tx
+        .inputs {
+            if *input.previous_output.is_coinbase {
+                let coinbase_block_height = *input.previous_output.block_height;
+                if block_height <= coinbase_block_height + 100 {
+                    errors
```

Comment:
Why not break on first error?

Reply:
ok

Reply:
i have the following 

```
    if !errors.is_empty() {
        return Result::Err( *errors[0]
            );        
    },
```
    
    `return Result::Err( *errors[0] ` throw the following error 
    
```
            return Result::Err( *errors[0]
                            ^********^
note: Trait has no implementation in context: core::traits::Copy::<core::byte_array::ByteArray>.
```
@maciejka what could be wrong here,

Reply:
> i have the following
> 
> ```
>     if !errors.is_empty() {
>         return Result::Err( *errors[0]
>             );        
>     },
> ```
> 
> ```
> `return Result::Err( *errors[0] ` throw the following error 
> ```
> 
> ```
> error: Cannot desnap a non copyable type.
>  --> /Users/macbookpro/Documents/onlydust/raito/src/validation/transaction.cairo:58:29
>         return Result::Err( *errors[0]
>                             ^********^
> note: Trait has no implementation in context: core::traits::Copy::<core::byte_array::ByteArray>..
> ```
> 
> @maciejka what could be wrong here,

fix this, by using the clone methond on` errors[0]`